src/nevermined/Assets.ts

```diff
@@ -345,6 +428,105 @@ export class Assets extends Instantiable {
         })
     }
 
+    public update(
+        did: string,
+        metadata: MetaData,
+        publisher: Account,
+        publishMetadata: PublishMetadata = PublishMetadata.JustMarketplaceAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<UpdateProgressStep, DDO> {
+
+        this.logger.log('Updating Asset')
+        return new SubscribablePromise(async observer => {
+            observer.next(UpdateProgressStep.ResolveAsset)
+            const ddo = await this.resolve(did)
+            
+
+            observer.next(UpdateProgressStep.UpdateMetadataInDDO)
+            const metadataService = ddo.findServiceByType('metadata')
+            
+            metadataService.attributes.additionalInformation = metadata.additionalInformation
+            metadataService.attributes.main = metadata.main
+            
+            await ddo.replaceService(
+                metadataService.index,
+                metadataService
+            )
+
+            observer.next(UpdateProgressStep.CalculateChecksum)
+            ddo.proof = await ddo.generateProof(publisher.getId())
+            const checksum = ddo.checksum(ddo.shortId())
+
+            observer.next(UpdateProgressStep.AddVersionInDDO)
+            
+            let lastIndex = 0
+            ddo._nvm.versions.map(v => v.id > lastIndex? lastIndex = v.id : false)
+            const ddoVersion: NvmConfigVersions = {
+                id: lastIndex + 1,
+                updated: new Date().toISOString().replace(/\.[0-9]{3}/, ''),
+                checksum: checksum,
+                immutableUrl: ''
```

Comment:
this can just be omitted

Reply:
since `immutableUrl` url is optional is there any point in including it if its just an empty string?

Reply:
removed